src/traversals/bfs.jl

```diff
@@ -170,3 +167,280 @@ function has_path(g::AbstractGraph{T}, u::Integer, v::Integer;
     end
     return false
 end
+
+using Base.Threads 
+
+"""
+    static_parallel_gdistances!(g, sources, vert_level; sort_alg=QuickSort)
+    static_parallel_gdistances!(g, source, vert_level; sort_alg=QuickSort)
+
+Parallel implementation of [`LightGraphs.gdistances!`](@ref) with static load balancing.
+
+### Performance
+Memory: `nthreads()*nv(g)*sizeof(eltype(g))`
+"""
+function static_parallel_gdistances!(
+    g::AbstractGraph{T}, 
+    sources::Vector{<:Integer},
+    vert_level::Vector{T};
+    sort_alg = Base.Sort.QuickSort
+    ) where T <:Integer
+ 
+    n = nv(g)
+    n_t = nthreads()
+
+    visited = zeros(Bool, n)
+    old_visited = zeros(Bool, n)
+#    visited = falses(n)
+#    old_visited = zeros(Bool, n)
+    fill!(vert_level, typemax(T))
+
+    # push! is not thread safe.
+    # Memory overhead could be reduced considerably if it were.
+    next_level_t = [Vector{T}(undef, n+1) for _ in 1:n_t]
+    vert_level = fill(typemax(T), n)
+    cur_level = Vector{T}(undef, length(sources))
+    cur_level .= sources
+    sizehint!(cur_level, n)
+
+    # End of queue is marked with 0
+    for t in 1:n_t
```

Comment:
Is this loop required as you are already setting the end of queue to zero on line 239?